File: app/src/main/java/me/evrooij/groceries/adapters/AccountAdapter.java
package me.evrooij.groceries.adapters;

import android.content.Context;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;
import me.evrooij.groceries.R;
import me.evrooij.groceries.domain.Account;

import java.util.ArrayList;

/**
 * Author: eddy
 * Date: 20-11-16.
 */

public class AccountAdapter extends ArrayAdapter<Account> {
    public AccountAdapter(Context context, ArrayList<Account> users) {
        super(context, 0, users);
    }

    @NonNull
    @Override
    public View getView(int position, View convertView, @NonNull ViewGroup parent) {
        // Get the data item for this position
        Account user = getItem(position);
        // Check if an existing view is being reused, otherwise inflate the view
        if (convertView == null) {
            convertView = LayoutInflater.from(getContext()).inflate(R.layout.item_user, parent, false);
        }
        // Lookup view for data population
        TextView tvName = (TextView) convertView.findViewById(R.id.tvName);
        // Populate the data into the template view using the data object
        if (user != null) {
            tvName.setText(user.getUsername());
        }

        // Return the completed view to render on screen
        return convertView;
    }
}

File: app/src/main/java/me/evrooij/groceries/ui/adapters/AccountAdapter.kt
package com.antonioleiva.weatherapp.ui.adapters

import android.support.v7.widget.RecyclerView
import android.util.SparseBooleanArray
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.antonioleiva.weatherapp.extensions.ctx
import kotlinx.android.synthetic.main.item_user.view.*
import me.evrooij.groceries.R
import me.evrooij.groceries.domain.Account
import me.evrooij.groceries.util.loadImg

class AccountAdapter(val accountList: List<Account>, val itemClick: (Account) -> Unit) : RecyclerView.Adapter<AccountAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.ctx).inflate(R.layout.item_user, parent, false)
        return ViewHolder(view, itemClick)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bindView(accountList[position])
    }

    override fun getItemCount() = accountList.size

    class ViewHolder(view: View, val itemClick: (Account) -> Unit) : RecyclerView.ViewHolder(view) {
        fun bindView(account: Account) {
            with(account) {
                itemView.ivProfilePicture.loadImg("")
                itemView.tvName.text = username
                itemView.setOnClickListener {
                    itemClick(this)
                }
            }
        }
    }
}